AsyncDisplayKit/Private/ASBatchFetching.m

```diff
@@ -37,19 +38,14 @@ BOOL ASDisplayShouldFetchBatchForContext(ASBatchContext *context,
     return NO;
   }
 
-  // Only Down and Right scrolls are currently supported (tail loading)
-  if (!ASScrollDirectionContainsDown(scrollDirection) && !ASScrollDirectionContainsRight(scrollDirection)) {
-    return NO;
-  }
-
   // No fetching for null states
-  if (leadingScreens <= 0.0 || CGRectEqualToRect(bounds, CGRectZero)) {
+  if (leadingScreens <= 0.0 || CGRectIsEmpty(bounds)) {
     return NO;
   }
 
   CGFloat viewLength, offset, contentLength;
 
-  if (ASScrollDirectionContainsDown(scrollDirection)) {
+  if (ASScrollDirectionContainsVerticalDirection(scrollableDirections)) {
```

Comment:
Not that it worked before, and I'm not sure how it would, but does this preclude batch fetching from working in collection views which scroll in both directions?